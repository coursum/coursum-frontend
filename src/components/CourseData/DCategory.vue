<template>
  <div
    v-if="
      courseDatas[idx].category && courseDatas[idx].category[`${curLang}`]
    "
    style="color: #929292"
    class="text-truncate text-caption"
  >
    {{ courseDatas[idx].category[`${curLang}`] }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { CourseInfo } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DCategory',
  props: {
    idx: {
      type: Number,
      default: 0,
    },
  },

  computed: {
    courseDatas(): CourseInfo[] {
      return this.$store.state.courseDatas;
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },

});
</script>
