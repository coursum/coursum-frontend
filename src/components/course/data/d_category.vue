<template>
  <div
    class="px-6 text-truncate"
  >
    {{ categoryData ? categoryData : $t("others") }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DCategory',
  props: {
    category: { type: Object, default: basicTemplate },
  },
  computed: {
    categoryData(): string | null | undefined {
      return this.category?.[this.curLang];
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
div{
  font-size: 0.7rem;
  color: gray;
}
</style>

<i18n>
{
  "en": {
    "others": "others"
  },
  "jp": {
    "others": "その他"
  }
}
</i18n>
