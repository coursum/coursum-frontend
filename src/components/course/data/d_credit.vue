<template>
  <v-chip
    v-if="credit"
    style="color: #929292"
    outlined
    x-small
  >
    {{ credit }}{{ $t("credit") }}
  </v-chip>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DCredit',
  props: {
    credit: {
      type: Number,
      default: undefined,
    },
  },
});
</script>

<i18n>
{
  "en": {
    "credit": "credit"
  },
  "jp": {
    "credit": "単位"
  }
}
</i18n>
