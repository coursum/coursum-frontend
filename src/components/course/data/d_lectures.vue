<template>
  <div
    v-if="lecturers && 0 < lecturers.length"
    :class="{'text-truncate': textTruncate}"
    class="px-6"
  >
    <template
      v-for="(lecturer, i) in lecturers"
    >
      <span
        v-if="lecturer.name && lecturer.name[`${curLang}`]"
        :key="i"
        style="display:inline-block;"
        class="mr-2"
      >
        {{ lecturer.name[`${curLang}`] }}
      </span>
    </template>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DLectures',
  props: {
    lecturers: {
      type: Array,
      default() {
        return undefined;
      },
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
span{
  font-size: 0.85rem;
  color: var(--v-wordcolor-base);
}
</style>
