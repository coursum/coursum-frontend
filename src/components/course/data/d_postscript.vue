<template>
  <span
    v-if="postscriptData"
    style="color: #929292"
    class="px-6 text-caption text-truncate"
  >
    {{ postscriptData }}
  </span>
</template>

<script lang="ts">
import { PropType, computed, defineComponent } from '@vue/composition-api';

import { Basic, basicTemplate } from '@/assets/CourseInfo';

export default defineComponent({
  name: 'DPostsctipt',
  props: {
    postscript: {
      type: Object as PropType<Basic>,
      default: basicTemplate,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'jp' | 'en');

    const postscriptData = computed((): string | null | undefined => props
      .postscript?.[curLang.value]);

    return {
      postscriptData,
    };
  },
});
</script>
