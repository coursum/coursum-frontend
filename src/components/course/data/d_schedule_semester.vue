<template>
  <v-chip v-if="semesterData"
          color="caption"
          outlined
          x-small
  >
    {{ semesterData }}
  </v-chip>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleSemester',
  props: {
    semester: {
      type: Object as PropType<Course['schedule']['semester']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');
    const semesterData = computed(() => props.semester[curLang.value]);

    return {
      semesterData,
    };
  },
});
</script>
