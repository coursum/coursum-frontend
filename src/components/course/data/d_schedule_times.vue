<template>
  <span v-if="courseTimes">
    <v-chip v-for="(time, i) in courseTimes" :key="i"
            color="caption"
            outlined
            x-small
    >
      {{ time }}
    </v-chip>
  </span>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleTimes',
  props: {
    times: {
      type: Object as PropType<Course['schedule']['times']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');

    const courseTimes = computed(() => {
      const times = props.times[curLang.value] || '';

      return times.split(',').map((time) => time.trim());
    });

    return {
      courseTimes,
    };
  },
});
</script>
