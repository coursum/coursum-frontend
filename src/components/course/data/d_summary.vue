<template>
  <div
    v-if="summaryData"
    :class="{summary: textTruncate}"
    class="px-6 primary--text py-1"
    :style="summaryStyle"
  >
    <p
      class="my-0"
    >
      {{ summaryData }}
    </p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DSummary',
  props: {
    summary: {
      type: Object,
      default: basicTemplate,
    },
    title: {
      type: Object,
      default: basicTemplate,
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return {
      summaryStyle: { 'font-size': '1rem' },
    };
  },
  computed: {
    summaryData(): string | null | undefined {
      if (this.titleJp === '研究会Ａ' || this.titleJp === '研究会Ｂ') {
        return this.summary?.en;
      }

      return this.summary?.[this.curLang];
    },
    titleJp(): string | null | undefined {
      return this.title?.jp;
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+JP&display=swap');

.summary p {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 5;
  overflow: hidden;
}
</style>
