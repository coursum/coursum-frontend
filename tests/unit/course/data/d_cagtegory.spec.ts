import { shallowMount } from '@vue/test-utils';

import { categoryMock } from '../../../data';
import { i18n } from '../../../setup';

import DCategory from '@/components/course/data/d_category.vue';

const correct = categoryMock.en;

const options = {
  i18n,
  propsData: {
    category: categoryMock,
  },
  mocks: {
    $t: (msg: string) => msg,
  },
};

describe('DCategory', () => {
  it('should render correct markup when prop is correct', () => {
    options.propsData.category.en = correct;
    const wrapper = shallowMount(DCategory, options);

    const div = wrapper.find('div');
    expect(div.text()).toEqual(correct);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correct markup when prop is falsy', () => {
    [undefined, null, ''].forEach((falsyProp: string | null | undefined) => {
      options.propsData.category.en = falsyProp;
      const wrapper = shallowMount(DCategory, options);

      const div = wrapper.find('div');
      expect(div.text()).toEqual('others');
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
