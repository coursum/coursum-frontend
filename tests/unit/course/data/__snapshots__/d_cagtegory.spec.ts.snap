// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DCategory should render correct markup when prop is correct 1`] = `
<div
  class="text-truncate"
>
  
  category en

</div>
`;

exports[`DCategory should render correct markup when prop is falsy 1`] = `
<div
  class="text-truncate"
>
  
  others

</div>
`;

exports[`DCategory should render correct markup when prop is falsy 2`] = `
<div
  class="text-truncate"
>
  
  others

</div>
`;

exports[`DCategory should render correct markup when prop is falsy 3`] = `
<div
  class="text-truncate"
>
  
  others

</div>
`;
